### DISCUSSION TITLE
With the streaming websocket API, will I get exactly one response per audio buffer submission?

### DISCUSSION THREAD

**Discussion by jsalsman** [deepgram/community#464] [General help] [answered]
If not, will I ever get zero? More than two? I'm wondering whether I use socket.recv() after each chunk of audio.

I'm using utterances=true if that makes a difference.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by jsalsman** [reply, marked as answer]
From David vonThenen on Discord:

> As all things in life, it depends. It mainly based on what your settings/options are. Utterance, endpointing, interim results, etc will all make a difference on how frequently you get a response.
> 
> If no one is speaking, you will definitely get an "empty" result because there is nothing to transcribe.
> 
> Ideally, you should have your send thread be running independently of the receive thread. For example, you can get multiple recv messages from a single send.

**Comment by davidvonthenen** [comment]
Details here:
https://discord.com/channels/1108042150941294664/1178896785347973120
